db/migrate/20170109115157_fix_lookup_key_auditable_type.rb

```diff
@@ -0,0 +1,15 @@
+class FixLookupKeyAuditableType < ActiveRecord::Migration
+  def up
+    lookup_key_ids = Audit.where(:auditable_type => 'LookupKey').pluck(:auditable_id).uniq
+
+    puppetclass_lookup_key_ids =  PuppetclassLookupKey.where(:id => lookup_key_ids).pluck(:id)
+    Audit.where(:auditable_type => 'LookupKey', :auditable_id => puppetclass_lookup_key_ids).update_all(:auditable_type => 'PuppetclassLookupKey')
```

Comment:
How about something like:
```ruby
Audit.joins('JOIN lookup_keys ON lookup_keys.id = audits.auditable_id').where(:auditable_type => 'LookupKey', :lookup_keys => {:type => 'VariableLookupKey'}).reorder(nil)
```
Also, make sure to `reorder(nil)` on all the queries to remove the default order scope.